src/api.ts

```
@@ -1,13 +1,47 @@
 import { MovieData } from './types/MovieData';
 import { ResponseError } from './types/ReponseError';
+import { Movie } from './types/Movie';
 
-const API_URL = 'https://www.omdbapi.com/?apikey=your-key';
+const API_URL = 'https://www.omdbapi.com/?apikey=932e507';
+
+export function getMovieInfo(imdbData: MovieData): Promise<Movie> {
+  return fetch(`${API_URL}&i=${imdbData.imdbID}`)
+    .then(res => {
+      if (!res.ok) {
+        throw new Error('Network error');
+      }
+
+      return res.json();
+    })
+    .then((data: MovieData) => ({
+      title: data.Title,
+      description: data.Plot,
+      imgUrl:
+        data.Poster === 'N/A'
+          ? 'https://via.placeholder.com/360x270.png?text=no%20preview'
+          : data.Poster,
+      imdbUrl: `https://www.imdb.com/title/${data.imdbID}`,
+      imdbId: data.imdbID,
+    }));
+}
 
 export function getMovie(query: string): Promise<MovieData | ResponseError> {
-  return fetch(`${API_URL}&t=${query}`)
-    .then(res => res.json())
+  return fetch(`${API_URL}&t=${encodeURIComponent(query)}`)
+    .then(res => {
+      if (!res.ok) {
+        throw new Error('Network error');
+      }
+
+      return res.json();
+    })
     .catch(() => ({
```

In `getMovie` you return a `ResponseError` object from `.catch(...)` which is good for signaling errors, but that swallow-all approach hides the original error details. Consider at least logging the original error (or attaching it) so debugging is easier. Also ensure consumers expect this shape (you already have `isError` guard).